test/gtest/uct/uct_test.cc

```diff
@@ -941,23 +943,27 @@ void uct_test::entity::mem_alloc_host(size_t length, unsigned mem_flags,
                              UCT_MEM_ALLOC_PARAM_FIELD_NAME;
     params.flags           = mem_flags;
     params.name            = "uct_test";
-    params.mem_type        = UCS_MEMORY_TYPE_HOST;
+    params.mem_type        = mem_type;
     params.address         = address;
 
-    if (md_attr().flags & (UCT_MD_FLAG_ALLOC|UCT_MD_FLAG_REG)) {
+    if ((md_attr().flags & (UCT_MD_FLAG_ALLOC | UCT_MD_FLAG_REG)) &&
+        (mem_type == UCS_MEMORY_TYPE_HOST)) {
         status = uct_iface_mem_alloc(m_iface, length, mem_flags, "uct_test",
                                      mem);
         ASSERT_UCS_OK(status);
     } else {
-        uct_alloc_method_t method = UCT_ALLOC_METHOD_MMAP;
-        status = uct_mem_alloc(length, &method, 1, &params, mem);
+        uct_alloc_method_t alloc_methods[] = {UCT_ALLOC_METHOD_MMAP,
+                                              UCT_ALLOC_METHOD_MD};
+        params.field_mask                 |= UCT_MEM_ALLOC_PARAM_FIELD_MDS;
+        params.mds.mds                     = &uct_md;
+        params.mds.count                   = 1;
+        status = uct_mem_alloc(length, alloc_methods, 1, &params, mem);
```

Comment:
You pass `num_methods` 1, with array of size 2.

Reply:
yep, can use ucs_static_array_size